execute/observation.go

```diff
@@ -359,7 +358,12 @@ func (p *Plugin) getFilterObservation(
 		}
 
 		for _, msg := range commitReport.Messages {
-			sender := typeconv.AddressBytesToString(msg.Sender[:], uint64(commitReport.SourceChain))
+			sender, err := p.addrCodec.AddressBytesToString(msg.Sender[:], commitReport.SourceChain)
+			if err != nil {
+				lggr.Errorw("unable to convert sender address to string", "err", err, "sender address", msg.Sender)
+				continue
```

Comment:
I would prefer to return error here, since something seriously wrong would cause the addresscodec to fail. By continuing, we are just silently ignoring the failure

@winder : Plz confirm

Reply:
We need to be careful about where errors are returned or a malicious NOP could crash the system by supplying bad observations.

Skipping over bad messages like this the usual approach.